src/framework/shortcuts/internal/shortcutsregister.cpp

```diff
@@ -256,9 +265,15 @@ Notification ShortcutsRegister::shortcutsChanged() const
     return m_shortcutsChanged;
 }
 
-const Shortcut& ShortcutsRegister::shortcut(const std::string& actionCode) const
+const Shortcut& ShortcutsRegister::shortcut(const std::string& actionCode)
 {
-    return findShortcut(m_shortcuts, actionCode);
+    const auto& shortCut = findShortcut(m_shortcuts, actionCode);
```

Comment:
This is a bit of a kludge, but it ensures that the ShortcutsRegister knows about all actions, even those that aren't in the shipped shortcuts.xml files, as currently the shortcuts dialog uses this to build the list of actions/shortcut assignments (but perhaps it should get them from UiActionsRegister instead...)

Reply:
BTW this can be removed from this PR easily enough, it was just to make it easier for me to test, but in the end it makes more sense to test by defining the shortcuts from MU3 and importing.  I would say the far better solution is to change the Shortcuts dialog to use the actions register to get the list of possible actions, then look up what any current shortcut assignments are using the shortcuts register.  I'm happy to start working on the preferences|shortcuts dialog as a separate PR if needed (there's quite a number of issues with it), that can be done independently of this one, but I've been tending to leave QML-related changes alone as I haven't managed to get qt creator working.